packages/core/src/env-set/create-pool-by-env.ts

```diff
@@ -0,0 +1,37 @@
+import { assertEnv } from '@silverhand/essentials';
+import inquirer from 'inquirer';
+import { createPool } from 'slonik';
+import { createInterceptors } from 'slonik-interceptor-preset';
+
+import appendDotEnv from './append-dot-env';
+
+const createPoolByEnv = async (isTest: boolean) => {
+  // Database connection is disabled in unit test environment
+  if (isTest) {
+    return;
+  }
+
+  const key = 'DB_URL';
+  const interceptors = [...createInterceptors()];
+
+  try {
+    const databaseDsn = assertEnv(key);
+
+    return createPool(databaseDsn, { interceptors });
+  } catch (error: unknown) {
+    const answer = await inquirer.prompt({
+      name: 'dsn',
+      message: `No Postgres DSN (${key}) found in env variables. Please input the DSN which points to Logto database:`,
+    });
+
+    if (!answer.dsn) {
```

Comment:
nit: do we need to validate the format of input dsn here

Reply:
Better trim and validate.
e.g. blank characters ` \t\r'` will not throw an error here.